kotlin/internal/klib/klib.bzl

```diff
@@ -10,8 +10,12 @@ def _kt_klib_library(ctx):
 
     toolchains = ctx.toolchains[_TOOLCHAIN_TYPE]
     deps_klibs = []
+    transitive_klibs = []
     for dep in ctx.attr.deps:
         deps_klibs.append(dep[_KtKlibInfo].klibs)
+        deps_klibs.append(dep[_KtKlibInfo].transitive_klibs)
```

Comment:
Does kotlin native support direct dependency compilation?

If not, just add the transitive deps to the `depset` `transitive` argument.